From 3977d628de9cf14b39237f4f3ddf5af41e42f102 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 03 Aug 2018 17:27:48 +0300
Subject: [PATCH] dali [flash tool]: add support for master partition

In case we define master partiotion of all device
skip it when programming all partitions

Also remove flash_erase since flashcp erase as well
---

diff --git a/board/habanalabs/common/rootfs_overlay/usr/sbin/flash_burn b/board/habanalabs/common/rootfs_overlay/usr/sbin/flash_burn
index 4551870..f58ec0f 100755
--- a/board/habanalabs/common/rootfs_overlay/usr/sbin/flash_burn
+++ b/board/habanalabs/common/rootfs_overlay/usr/sbin/flash_burn
@@ -56,6 +56,9 @@
 	BLK_SIZE=`mtdinfo ${DEV}	| grep "Eraseblock size"	| tr -s ' ' | cut -d: -f2	| cut -d' ' -f2`
 	BLK_CNT=`mtdinfo ${DEV}		| grep "Amount of eraseblocks"	| tr -s ' ' | cut -d: -f2	| cut -d' ' -f2`
 	DEVID=`mtdinfo ${DEV}		| grep "major/minor"		| tr -s ' ' | cut -d: -f2-3	| cut -d' ' -f2`
+	if [ ! -e /sys/dev/char/${DEVID}/offset ]; then
+		return
+	fi
 	OFFSET=`cat /sys/dev/char/${DEVID}/offset`
 	OFFSET_HEX=`printf "0x%x" ${OFFSET}`
 	SKIP_CNT=$((${OFFSET}/${BLK_SIZE}))
@@ -63,7 +66,7 @@
 
 	dd status=none conv=sync bs=${BLK_SIZE} skip=${SKIP_CNT} if=${INFILE} of=${IMG} count=${BLK_CNT}
 	echo -n "    ${NAME} @ ${OFFSET_HEX}... "
-	flash_erase -q -u ${DEV} 0 ${BLK_CNT}
+	#flash_erase -q -u ${DEV} 0 ${BLK_CNT}
 	flashcp ${IMG} ${DEV}
 
 	dd status=none conv=sync bs=${BLK_SIZE} if=${DEV} of=${IMG}.verify count=${BLK_CNT}
